### PR TITLE
add alternate file

### PR DESCRIPTION
**Pull request by robinvd**
inspired by vim ctrl-6/kak ga commands. the alternate file is kept per view

### CODE REVIEW

File: helix-view/src/editor.rs [hunk 1]
```diff
@@ -88,6 +88,12 @@ impl Editor {
     pub fn switch(&mut self, id: DocumentId, action: Action) {
         use crate::tree::Layout;
         use helix_core::Selection;
+
+        if !self.documents.contains_key(id) {
+            log::warn!("cannot switch to document that does not exist (anymore)");
```
Comment (archseer) [inline helix-view/src/editor.rs:93]: Let's make this `log::error!`, ideally this shouldn't happen. We'll need to add something like `forget_doc` on the view to remove it from both alternate file and the jumplist: https://github.com/mawww/kakoune/blob/c507863a00289fe4b210ecc4fef5554de4c87bf4/src/context.cc#L197-L210

File: helix-view/src/view.rs [hunk 2]
```diff
@@ -67,6 +67,7 @@ pub struct View {
     pub first_col: usize,
     pub area: Rect,
     pub jumps: JumpList,
+    pub alternate_file: Option<DocumentId>,
```
Comment (archseer) [inline helix-view/src/view.rs:70]: Let's call this `alternate_doc` to match the `doc` field.

**Review by archseer — CHANGES_REQUESTED** [review]
Thanks, I've been missing this feature! Just a few minor comments.

**Comment by pickfire** [comment]
I wish it could be like vim `^`, `ga` is not very useful for me since I always browse file, maybe I will port the patch I sent for kakaoune https://github.com/mawww/kakoune/pull/3814 here. For last modified file which behaves more like vim. So browsing file don't need to override it.

**Comment by robinvd** [comment]
@pickfire i think that would be a great addition as well

**Comment by pickfire** [comment]
@robinvd Can you change those `alternate` to `last_accessed` to be clearer so stuff will be easier once I add the other one?

**Comment by robinvd** [comment]
@pickfire  done

**Review by pickfire — APPROVED** [review]
Looks good to me. Not tested but I will test later.

**Comment by pickfire** [comment]
Tested, but I think it would be better to have a text in the status line saying "no last accessed file" when `ga` is pressed without last accessed file.

**Review by archseer — APPROVED** [review]

**Pull request merged**